### ISSUE TITLE
[Feature]: Custom boundary for autoSize in Popper

### ISSUE THREAD

**Issue by miroslavstastny** [microsoft/fluentui#25264]
### Library

React Northstar / v0 (@fluentui/react-northstar)

### Describe the feature that you would like added

`autoSize` prop in Popper settings currently uses `overflowBoundary`. Add configuration option for the `autoSize` to use a separate boundary.

That feature is required for a `Dropdown` component scenario.
Codesandbox with repro: https://codesandbox.io/s/v0-dropdown-overflow-vj5xzf?file=/example.js

The `Dropdown` list in the example has the following options applied:
`positionFixed: true` to allow the dropdown list to overflow the message it is rendered in
`autoSize: "width-always"` to fix the width of the list to match the dropdown input (required when `fluid` and `positionFixed` are used at the same time)
`overflowBoundary` and `flipBoundary` set to the scroll container to correctly flip the dropdown list
`unstable_disableTether: "all"` to allow the dropdown list to cover the input if there is not enough space

With this config the `autoSize` uses the `overflowBoundary` to set the dropdown list width which is incorrect:
![image](https://user-images.githubusercontent.com/9615899/196358400-48f5eb98-7964-435a-ad73-f486abff4ce1.png)

### Validations

- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

**Comment by miroslavstastny** [comment]
We should test and eventually fix in v9 first, only after that decide what to do next.

**Comment by JustSlone** [comment]
Agree, let's solve this in v9, then backport to v0 based on customer need / cost tradeoff. 

I assume @microsoft/teams-prg is taking point given the assignment in board, correct if wrong. 

CC: @smhigley and @microsoft/cxe-coastal

**Comment by layershifter** [comment]
unclose

**Comment by layershifter** [comment]
keep open

**Comment by layershifter** [comment]
keep open
